network/network.go

```diff
@@ -1126,6 +1130,23 @@ func (n *network) dial(ctx context.Context, nodeID ids.NodeID, ip *trackedIP) {
 				n.config.MaxReconnectDelay,
 			)
 
+			// If the network is configured to disallow private IPs and the
+			// provided IP is private, we skip all attempts to initiate a
+			// connection.
+			//
+			// Invariant: We perform this check inside of the looping goroutine
+			// because this goroutine must clean up the trackedIPs entry if
+			// nodeID leaves the validator set.
+			if !n.config.AllowPrivateIPs && ip.ip.IP.IsPrivate() {
+				n.peerConfig.Log.Verbo("skipping connection dial",
+					zap.String("reason", "outbound connections to private IPs are prohibited"),
+					zap.Stringer("nodeID", nodeID),
+					zap.Stringer("peerIP", ip.ip.IP),
+					zap.Duration("delay", ip.delay),
+				)
+				continue
```

Comment:
from comment `we skip all attempts to initiate a connection`. 
aren't we actually just attempting again here with `continue`?  should we return it here? 
If we meant to keep this so we can loop it over and clean trackedIps, can we clarify on comments why we use continue instead of return? that "skip all attempts" could be confusing.

Reply:
I added some additional context in the invariant comment and modified a test so that it will fail if this is changed to `return`.